ml-agents/mlagents/trainers/trajectory.py

```diff
@@ -0,0 +1,128 @@
+from typing import List, NamedTuple
+import numpy as np
+
+from mlagents.trainers.buffer import AgentBuffer
+
+
+class AgentExperience(NamedTuple):
+    obs: List[np.ndarray]
+    reward: float
+    done: bool
+    action: np.ndarray
+    action_probs: np.ndarray
+    action_pre: np.ndarray  # TODO: Remove this
+    action_mask: np.ndarray
+    prev_action: np.ndarray
+    max_step: bool
+    memory: np.ndarray
+
+
+class SplitObservations(NamedTuple):
+    vector_observations: np.ndarray
+    visual_observations: List[np.ndarray]
+
+    @staticmethod
+    def from_observations(obs: List[np.ndarray]) -> "SplitObservations":
+        """
+        Divides a List of numpy arrays into a SplitObservations NamedTuple.
+        This allows you to access the vector and visual observations directly,
+        without enumerating the list over and over.
+        :param obs: List of numpy arrays (observation)
+        :returns: A SplitObservations object.
+        """
+        vis_obs_indices = []
```

Comment:
I think you should avoid creating the indices arrays here. You can form `vis_obs` and the input to `np.concatenate()` directly on the loop through `obs`.

Reply:
Good call - I now just build the list directly. 

I think we use the same logic in the `brain_conversion_utils.py` file as well